### ISSUE TITLE
The single_value function throws an error when missing data in referenced store

### ISSUE THREAD

**Issue by ollien1** [atoti/atoti#702]
atoti 0.7.0

When used on a column in a reference store, the single value function can throw an error if there are rows in the base store that do not map onto any rows in the reference store.

Code to reproduce:

```py
import atoti as tt
import pandas as pd
from datetime import date

session = tt.Session()

base = session.read_pandas(
    pd.DataFrame(
        columns=["Date", "Id", "RefId", "Misc", "Qty"],
        data=[
            (date.fromisoformat("2022-10-03"), "1", "A", "X", 5),
            (date.fromisoformat("2022-10-03"), "2", "A", "X", 10),
            (date.fromisoformat("2022-10-03"), "3", "B", "X", 20),
            (date.fromisoformat("2022-10-03"), "4", "B", "Y", 30),
            (date.fromisoformat("2022-10-03"), "5", "B", "Y", 25),
        ],
    ),
    table_name="Base",
    keys=["Date", "Id"],
    types={"Date": tt.type.LOCAL_DATE}
)

ref = session.read_pandas(
    pd.DataFrame(
        columns=["Date", "RefId", "Rate"],
        data=[
            (date.fromisoformat("2022-10-03"), "A", 10)
        ],
    ),
    table_name="Ref",
    keys=["Date", "RefId"],
    types={"Date": tt.type.LOCAL_DATE}
)

base.join(ref)

cube = session.create_cube(base)

m = cube.measures

m["Rate"] = tt.agg.single_value(ref["Rate"])

cube.query(m["Rate"])
```

Note I attempted to recreate this with only a single key field in the reference store (no date column) and the issue did not occur.

**Comment by mailys-ds** [comment]
Hi !
Thanks for reporting this issue, I can reproduce it. We'll work on a fix.

**Comment by bloussou** [comment]
Hello @ollien1 ,
This will be fixed in the next release (`0.7.1`).
